src/Playwright/Core/RouteHandler.cs

```diff
@@ -37,9 +36,7 @@ internal class RouteHandler
 
     private bool _ignoreException;
 
-    public Regex Regex { get; set; }
-
-    public Func<string, bool> Function { get; set; }
+    public URLMatch URL { get; set; }
```

Comment:
I'd rename URL to `urlMatcher` and  also drop `*Match` suffixes in reMatch, globMatch and funcMatch.